packages/cli/src/lib/snapshot/snapshotReporter.ts

```diff
@@ -197,25 +186,22 @@ export class SnapshotReporter {
 
   private addResourceInError(
     resourceType: ResourceSnapshotType,
-    // TODO CDX-1038: Clean that.
-    err: string | ResourceSnapshotsReportOperationResult
+    err: ResourceSnapshotsReportOperationResult
   ) {
     this.resourceInErrorCount++;
     let errorSet = this.resourceInError.get(resourceType);
     if (!errorSet) {
       errorSet = new Set();
       this.resourceInError.set(resourceType, errorSet);
     }
-    // TODO CDX-1038: Clean that.
-    errorSet.add(typeof err === 'string' ? err : err.message);
+    errorSet.add(`${err.resultCode}: ${err.message}`);
```

Comment:
Should we include the resultCode in the error string?
I think it would not hurt